packages/calcite-components/src/components/autocomplete/autocomplete.tsx

```diff
@@ -0,0 +1,849 @@
+import { PropertyValues } from "lit";
+import {
+  LitElement,
+  property,
+  createEvent,
+  h,
+  method,
+  state,
+  JsxNode,
+  stringOrBoolean,
+} from "@arcgis/lumina";
+import { useWatchAttributes } from "@arcgis/components-controllers";
+import { debounce } from "lodash-es";
+import {
+  FlipPlacement,
+  FloatingCSS,
+  FloatingLayout,
+  FloatingUIComponent,
+  MenuPlacement,
+  OverlayPositioning,
+  connectFloatingUI,
+  defaultMenuPlacement,
+  disconnectFloatingUI,
+  reposition,
+} from "../../utils/floating-ui";
+import {
+  InteractiveComponent,
+  InteractiveContainer,
+  updateHostInteraction,
+} from "../../utils/interactive";
+import {
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent,
+} from "../../utils/loadable";
+import { onToggleOpenCloseComponent, OpenCloseComponent } from "../../utils/openCloseComponent";
+import { Alignment, Scale, Status } from "../interfaces";
+import { IconNameOrString } from "../icon/interfaces";
+import { connectLabel, disconnectLabel, LabelableComponent } from "../../utils/label";
+import { TextualInputComponent } from "../input/common/input";
+import {
+  afterConnectDefaultValueSet,
+  FormComponent,
+  HiddenFormInputSlot,
+  MutableValidityState,
+  connectForm,
+  disconnectForm,
+  submitForm,
+} from "../../utils/form";
+import { slotChangeHasAssignedElement } from "../../utils/dom";
+import { guid } from "../../utils/guid";
+import { useT9n } from "../../controllers/useT9n";
+import type { Input } from "../input/input";
+import type { AutocompleteItem } from "../autocomplete-item/autocomplete-item";
+import type { AutocompleteItemGroup } from "../autocomplete-item-group/autocomplete-item-group";
+import type { Label } from "../label/label";
+import { Validation } from "../functional/Validation";
+import { createObserver } from "../../utils/observers";
+import { styles } from "./autocomplete.scss";
+import T9nStrings from "./assets/t9n/autocomplete.t9n.en.json";
+import { CSS, ICONS, IDS, SLOTS } from "./resources";
+
+const groupItemSelector = "calcite-autocomplete-item-group";
+const itemSelector = "calcite-autocomplete-item";
+
+declare global {
+  interface DeclareElements {
+    "calcite-autocomplete": Autocomplete;
+  }
+}
+
+/**
+ * @slot - A slot for adding `calcite-autocomplete-item` elements.
+ * @slot actions-end - A slot for adding actions after the input.
+ * @slot actions-start - A slot for adding actions before the input.
```

Comment:
@ashetland @SkyeSeitz I wonder if these need to be part of this component or if this is really just layout consideration for the implementer?

Reply:
I believe the action slots where an mvp requirement, but I'm unsure about the reasons. Agree they could be just a layout consideration and I'm not sure if the slots are helping much given the differences in Button and Action styling.

Reply:
Yeah, maybe we can look at that in the future, in Input too. I believe the intent there, years ago, was to support some kind of built in submit button developers wired up. But if all the slots are doing are putting content next to the component, maybe that's out of the component's scope and should just be a styling / layout consideration of the implementer.

Reply:
It makes it a little more difficult on the end user to have a button next to the component. They would have to setup their own flex container to do it. If we had utility classes or components it would be much easier.

I could go either way on this one.

Reply:
I can remove them if we think input will have its removed at some point in the future.

Reply:
I'd think any change there is a ways out - we could just address it then. Agreed on needing some layout helpers for making layout and consistent spacing of components a bit easier.

Reply:
Removed these slots.